### PR TITLE
feat(split): small refactor and fix of replica split

### PR DESCRIPTION
**Pull request by hycdong**
This pull request is about `replica_split_manager`, including:
- update some log and comments
- update rpc call

### CODE REVIEW

File: src/replica/split/replica_split_manager.cpp [hunk 1]
```diff
@@ -781,6 +785,11 @@ void replica_split_manager::on_register_child_on_meta_reply(
 void replica_split_manager::child_partition_active(
     const partition_configuration &config) // on child
 {
+    if (status() != partition_status::PS_PARTITION_SPLIT) {
+        dwarn_replica("child partition has been active, status={}", enum_to_string(status()));
+        return;
+    }
+
     ddebug_replica("child partition become active");
```
Comment (neverchanje) [inline src/replica/split/replica_split_manager.cpp:793]:
```suggestion
    if (status() != partition_status::PS_PARTITION_SPLIT) {
        dwarn_replica("child partition has been active, status={}", enum_to_string(status()));
        return;
    }

    ddebug_replica("child partition become active");
```

The two logs seem ambiguous. They both say the partition is "active". I think you should add a comment here, stating what it means "active". And I guess the latter log should be "is becoming active".

File: src/replica/split/replica_split_manager.cpp [hunk 2]
```diff
@@ -199,13 +198,14 @@ void replica_split_manager::parent_prepare_states(const std::string &dir) // on
...
     plist->truncate(last_committed_decree());
 
-    dassert_replica(
-        last_committed_decree() == checkpoint_decree || !mutation_list.empty() || !files.empty(),
-        "last_committed_decree({}) VS checkpoint_decree({}), mutation_list size={}, files size={}",
-        last_committed_decree(),
-        checkpoint_decree,
-        mutation_list.size(),
-        files.size());
+    dassert_replica(last_committed_decree() == checkpoint_decree || !mutation_list.empty() ||
```
Comment (neverchanje) [inline src/replica/split/replica_split_manager.cpp:201]:
```suggestion
    dcheck_eq(last_committed_decree(), checkpoint_decree);
    dcheck_gt(mutation_list.size(), 0);
    dcheck_gt(files.size(), 0);
```

File: src/replica/split/replica_split_manager.cpp [hunk 3]
```diff
@@ -509,39 +505,41 @@ void replica_split_manager::child_notify_catch_up() // on child partition
...
-                     return;
-                 }
-                 ddebug_replica("notify primary catch up succeed");
-             },
-             _replica->_split_states.parent_gpid.thread_hash());
+    notify_catch_up_rpc rpc(std::move(request),
+                            RPC_SPLIT_NOTIFY_CATCH_UP,
+                            0_ms,
+                            0,
+                            _replica->_split_states.parent_gpid.thread_hash());
```
Comment (neverchanje) [inline src/replica/split/replica_split_manager.cpp:517]:
```suggestion
    notify_catch_up_rpc rpc(std::move(request),
                            RPC_SPLIT_NOTIFY_CATCH_UP,
                            /*never timeout*/0_ms,
                            /*partition_hash*/0,
                            _replica->_split_states.parent_gpid.thread_hash());
```

File: src/replica/split/replica_split_manager.cpp [hunk 4]
```diff
@@ -692,19 +696,18 @@ void replica_split_manager::parent_send_register_request(
 
     rpc_address meta_address(_stub->_failure_detector->get_servers());
     std::unique_ptr<register_child_request> req = make_unique<register_child_request>(request);
-    register_child_rpc rpc(std::move(req), RPC_CM_REGISTER_CHILD_REPLICA);
+    register_child_rpc rpc(
+        std::move(req), RPC_CM_REGISTER_CHILD_REPLICA, 0_ms, 0, get_gpid().thread_hash());
```
Comment (neverchanje) [inline src/replica/split/replica_split_manager.cpp:700]:
```suggestion
    register_child_rpc rpc(
        std::move(req), RPC_CM_REGISTER_CHILD_REPLICA, /*never timeout*/ 0_ms, /*partition hash*/ 0, get_gpid().thread_hash());
```

File: src/replica/split/replica_split_manager.cpp (hunk 1 above)
Comment (hycdong) [reply src/replica/split/replica_split_manager.cpp]: Yes, I move the latter log after child status changed, it will be clearer.

**Review by acelyc111 — APPROVED** [review]

**Review by Shuo-Jia — APPROVED** [review]

**Pull request merged**